artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java

```diff
@@ -1075,17 +1089,24 @@ void stop(boolean failoverOnServerShutdown, final boolean criticalIOError, boole
          state = SERVER_STATE.STOPPING;
 
          if (criticalIOError) {
-            // notifications trigger disk IO so we don't want to send any on a critical IO error
-            managementService.enableNotifications(false);
+            final ManagementService managementService = this.managementService;
```

Comment:
why do we need this variable? who is setting to null this variable on stopping?

Reply:
It's thanks to Java Language Specification: the local `managementService` cannot change once read